src/ducks/wallet/ledger.ts

```diff
@@ -0,0 +1,70 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { ActionStatus, RejectMessage } from "constants/types.d";
+import LedgerTransport from "@ledgerhq/hw-transport-u2f";
+import LedgerApi from "@ledgerhq/hw-app-str";
+
+export const fetchLedgerStellarAddressAction = createAsyncThunk<
+  { publicKey: string },
+  string,
+  { rejectValue: RejectMessage }
+>(
+  "walletLedger/fetchLedgerStellarAddressAction",
+  async (bipPath, { rejectWithValue }) => {
+    const result = { publicKey: "" };
+    try {
+      const transport = await LedgerTransport.create();
+      const ledgerApi = new LedgerApi(transport);
+      const response = await ledgerApi.getPublicKey(bipPath);
+      result.publicKey = response.publicKey;
+    } catch (error) {
+      return rejectWithValue({ errorString: error.toString() });
+    }
+    return result;
+  },
+);
+
+interface InitialState {
+  data: null | { publicKey: string };
```

Comment:
I see that both Trezor and Ledger data types are the same, we could create a type for it. What do you think?

Reply:
Ya that's a good idea. I'll make it **WalletInitialState**